streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverSchedulingPolicy.scala

```diff
@@ -144,27 +141,31 @@ private[streaming] class ReceiverSchedulingPolicy {
     val scheduledExecutors = mutable.Set[String]()
     scheduledExecutors ++= preferredLocation
 
-    val executorWeights = receiverTrackingInfoMap.values.flatMap { receiverTrackingInfo =>
-      receiverTrackingInfo.state match {
-        case ReceiverState.INACTIVE => Nil
-        case ReceiverState.SCHEDULED =>
-          val scheduledExecutors = receiverTrackingInfo.scheduledExecutors.get
-          // The probability that a scheduled receiver will run in an executor is
-          // 1.0 / scheduledLocations.size
-          scheduledExecutors.map(location => location -> (1.0 / scheduledExecutors.size))
-        case ReceiverState.ACTIVE => Seq(receiverTrackingInfo.runningExecutor.get -> 1.0)
-      }
+    val executorWeights = receiverTrackingInfoMap.filter(_._1 != receiverId).values.flatMap {
```

Comment:
Updated to use the new approach. And since we don't call rescheduleReceiver when the receiver registers at the first time, it's not necessary to do filtering.